langchain/tracing/nested_json.py

```diff
@@ -0,0 +1,147 @@
+"""An implementation of the Tracer interface that prints trace as nested json."""
+
+import datetime
+import uuid
+from typing import Any, Dict, List, Optional, Union
+
+from langchain.tracing.base import (
+    BaseTracer,
+    ChainRun,
+    LLMRun,
+    ToolRun,
+    TracerException,
+)
+
+
+class NestedJsonTracer(BaseTracer):
+    """An implementation of the Tracer interface that prints trace as nested json."""
+
+    _instance = None
+
+    def __new__(cls):
```

Comment:
The idea is that to avoid having to pass the logger through calls in a chain, we need a global shared logger, i.e. a singleton. To prevent the multiple instantiations of this class, we whether a **class** was instantiated. That's why we use `__new__`... it takes in the class (`cls`) to control object creation where as `__init__`'s job is to take in the instance and handle setting instance variables. 

Good point though, I'll add a comment